tensorboard/components/tf_ng_tensorboard/plugins/plugins.component.ts

```diff
@@ -57,8 +74,28 @@ export class PluginsComponent implements OnInit {
     // We manually create plugin DOM (with custom tagName and script inside
     // an iframe) when the `activePlugin` changes.
     this.activePlugin$
-      .pipe(filter(Boolean))
-      .subscribe((plugin) => this.renderPlugin(plugin as UiPluginMetadata));
+      .pipe(
+        filter(Boolean),
+        distinctUntilChanged(
+          (prev: UiPluginMetadata, curr: UiPluginMetadata) =>
+            prev.id === curr.id
+        )
+      )
+      .subscribe((plugin: UiPluginMetadata) => this.renderPlugin(plugin));
```

Comment:
I am aware except I was not relying on Angular to do the plugin rendering for me since they are polymer components.

Of course, you can argue that a thin `component` that implements `OnChange` is simpler than using observable. Is that the fair representation of your comment here?

Reply:
Yep that's it :)